### PR TITLE
Update chemfiles tests for 0.10.3

### LINKED ISSUE #3798: `Chemfiles`  change in float writing for 0.10.3 causing CI failures
There seems to be an issue with `chemfiles` that is causing the full-deps builds to fail in #3127 and #3777 (likely others as well). I have re-triggered a few other jobs to check. 

It is worth noting that `chemfiles` released a new version on [pip](https://pypi.org/project/chemfiles/0.10.3/) yesterday (0.10.3) which could be the issue [see here for the tag](https://github.com/chemfiles/chemfiles.py/releases/tag/0.10.3). 

 I could reproduce the error (which seemed to be related to finding some shared library, at least on my computer) locally with a conda install of `chemfiles` but seemed to work OK for a pip install.

### PR DESCRIPTION
**Pull request by Luthaf**
Fixes #3798

Changes made in this Pull Request:
 - Updated the tests for chemfiles 0.10.3, skipping them for lower versions.

PR Checklist
------------
 - [X] Tests?
 - [ ] ~Docs?~
 - [ ] ~CHANGELOG updated?~
 - [X] Issue raised/referenced?

### CODE REVIEW

**Review by hmacdope — APPROVED** [review]
Looks good to me @Luthaf. I would just add a `CHANGELOG` entry.

File: testsuite/MDAnalysisTests/coordinates/test_chemfiles.py
Comment (IAlibay) [inline testsuite/MDAnalysisTests/coordinates/test_chemfiles.py]: This really isn't ideal, if anything we really shouldn't have code that does an indirect explicit float comparison like this.

The main question is - was this change in writing behaviour intended?

File: testsuite/MDAnalysisTests/coordinates/test_chemfiles.py
Comment (IAlibay) [inline testsuite/MDAnalysisTests/coordinates/test_chemfiles.py:47]: As far as I can tell (please correct me if I'm right) these ' to " don't adhere to any given convention, why are they all being changed? (i.e. is this worth the git history  footprint)

**Review by IAlibay — CHANGES_REQUESTED** [review]
I'm going to block for details - essentially I'd like to make sure this isn't just a temporary patch to match some arbitrary change in float writing.

File: testsuite/MDAnalysisTests/coordinates/test_chemfiles.py
Comment (RMeli) [reply testsuite/MDAnalysisTests/coordinates/test_chemfiles.py:47]: It's `black`'s convention: https://black.readthedocs.io/en/stable/the_black_code_style/current_style.html#strings

Comment (Luthaf) [reply testsuite/MDAnalysisTests/coordinates/test_chemfiles.py:47]: Exactly, these were automated changes by black. I can revert them if you prefer.

File: testsuite/MDAnalysisTests/coordinates/test_chemfiles.py
Comment (Luthaf) [reply testsuite/MDAnalysisTests/coordinates/test_chemfiles.py]: Yes, the change was intended on chemfiles side. I can use a different format (maybe a binary format?) than LAMMS data files to protect such small changes, since the purpose of this tests it to check velocities are passed & used by chemfiles; not the actual output.

Comment (hmacdope) [reply testsuite/MDAnalysisTests/coordinates/test_chemfiles.py]: IIRC its closer to how ASE writes data files but not LAMMPs itself does, but moot point anyway as this is about chemfiles support.

Comment (IAlibay) [reply testsuite/MDAnalysisTests/coordinates/test_chemfiles.py]: As long as it's an intended change in writing behaviour, and that chemfiles can guarantee there won't be any arbitrary (i.e. non-intentional) changes in this writing behaviour, then I'm happy.

I know I'm being a pedantic here, but it's better being clear about how things are intended to work now than to have to come back here in 3-6 months.

File: testsuite/MDAnalysisTests/coordinates/test_chemfiles.py
Comment (IAlibay) [reply testsuite/MDAnalysisTests/coordinates/test_chemfiles.py:47]: I'm not a fan of arbitrary (read non-PEP/NEP/*EP) line changes, especially if they affect the history. However, I will leave it to you to decide on what to do here.

File: testsuite/MDAnalysisTests/coordinates/test_chemfiles.py
Comment (Luthaf) [reply testsuite/MDAnalysisTests/coordinates/test_chemfiles.py]: I totally understand wanting to be clear! We have very similar tests on the chemfiles side (e.g. https://github.com/chemfiles/chemfiles/blob/387d4920b7eb5107ccd6050da238f82dacf1c85b/tests/formats/lammps-data.cpp#L115), so there should not be any unintentional changes here.

There might still be intentional changes that would introduce extra "noise" on the MDA side (although I can be more mindful of this), so I'll still try to write a more stable test here, maybe using a different format.

File: testsuite/MDAnalysisTests/coordinates/test_chemfiles.py
Comment (Luthaf) [reply testsuite/MDAnalysisTests/coordinates/test_chemfiles.py:47]: I tend to treat `black` as mostly standard nowadays, especially given it is developed by the PSF, and the output is quite stable.

File: testsuite/MDAnalysisTests/coordinates/test_chemfiles.py
Comment (IAlibay) [reply testsuite/MDAnalysisTests/coordinates/test_chemfiles.py]: Extra noise isn't as much of an issue (to me at least) to be honest (although more stable tests are always appreciated, that can be future work if you want).

File: testsuite/MDAnalysisTests/coordinates/test_chemfiles.py
Comment (IAlibay) [reply testsuite/MDAnalysisTests/coordinates/test_chemfiles.py:47]: Actually had a long discussion yesterday with @RMeli (or more we have the same long discussion every couple of months).

My stance is a pretty simple yet strong one re: `black`'s "style": if the PSF wants to codify it as a PEP or even a BEP, then I'd be happy adopting it & evangelizing it. Otherwise it remains a set of decisions set by an arbitrary tool choice, which makes it hard for me to go "no you should do X not Y".

In any case, given you're the code author of the lines being changed, not a hill I wish to spend more time on.

**Review by IAlibay — APPROVED** [review]
I'll approve early here - if you could just add yourself to the changelog for the 2.4.0 release that'd be great!

File: testsuite/MDAnalysisTests/coordinates/test_chemfiles.py
Comment (Luthaf) [reply testsuite/MDAnalysisTests/coordinates/test_chemfiles.py]: I've updated the test to use a NetCDF file instead, which can easily be checked with scipy. This should be much more stable (up to the byte!) now

**Comment by Luthaf** [comment]
CHANGELOG done, I tried to follow the previous release template

File: package/CHANGELOG
Comment (IAlibay) [inline package/CHANGELOG]:
```suggestion
  * Upgrade to chemfiles 0.10.3 (Issue #3798)
```

**Review by IAlibay — APPROVED** [review]
Just the one thing

**Pull request merged**
